### ISSUE TITLE
gauge dotnet (and csharp) templates do not work on Windows 10

### ISSUE THREAD

**Issue by kuechlerm** [getgauge/gauge#1125]
### Expected behavior
gauge csharp and dotnet templates should work.

### Actual behavior
Specs do not run. Output:

```
Microsoft (R)-Buildmodul, Version 15.7.179.6572 f�r .NET Core
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

# Specification Heading
  ## Vowel counts in single word        Assert Failure
Expression: [Recursive resource lookup bug]
Description: Infinite recursion during resource lookup within System.Private.CoreLib.  This may be a bug in System.Private.CoreLib, or potentially in certain extensibility points such as assembly resolve events or CultureInfo names.  Resource name: ArgumentNull_Generic
Stack Trace:
   at System.SR.InternalGetResourceString(String key)
   at System.SR.GetResourceString(String resourceKey, String defaultString)
   at System.ArgumentNullException..ctor(String paramName)
   at System.Runtime.Loader.AssemblyLoadContext.GetLoadContext(Assembly assembly)
   at System.Reflection.Assembly.LoadFromResolveHandler(Object sender, ResolveEventArgs args)
   at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks, IntPtr ptrLoadContextBinder)
   at System.Reflection.RuntimeAssembly.InternalGetSatelliteAssembly(String name, CultureInfo culture, Version version, Boolean throwOnFileNotFound, StackCrawlMark& stackMark)
   at System.Resources.ManifestBasedResourceGroveler.GetSatelliteAssembly(CultureInfo lookForCulture, StackCrawlMark& stackMark)
   at System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo culture, Dictionary`2 localResourceSets, Boolean tryParents, Boolean createIfNotExists, StackCrawlMark& stackMark)
   at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo requestedCulture, Boolean createIfNotExists, Boolean tryParents, StackCrawlMark& stackMark)
   at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents)
   at System.Resources.ResourceManager.GetString(String name, CultureInfo culture)
   at System.SR.InternalGetResourceString(String key)
   at System.SR.GetResourceString(String resourceKey, String defaultString)
   at System.ArgumentNullException..ctor(String paramName)
   at System.Runtime.Loader.AssemblyLoadContext.GetLoadContext(Assembly assembly)
   at System.Reflection.Assembly.LoadFromResolveHandler(Object sender, ResolveEventArgs args)
   at System.AppDomain.OnAssemblyResolveEvent(RuntimeAssembly assembly, String assemblyFullName)
   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks, IntPtr ptrLoadContextBinder)
   at System.Reflection.RuntimeAssembly.InternalGetSatelliteAssembly(String name, CultureInfo culture, Version version, Boolean throwOnFileNotFound, StackCrawlMark& stackMark)
   at System.Resources.ManifestBasedResourceGroveler.GetSatelliteAssembly(CultureInfo lookForCulture, StackCrawlMark& stackMark)
   at System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo culture, Dictionary`2 localResourceSets, Boolean tryParents, Boolean createIfNotExists, StackCrawlMark& stackMark)
   at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo requestedCulture, Boolean createIfNotExists, Boolean tryParents, StackCrawlMark& stackMark)
   at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents)
   at System.Resources.ResourceManager.GetString(String name, CultureInfo culture)
   at System.SR.InternalGetResourceString(String key)
   at System.SR.GetResourceString(String resourceKey, String defaultString)
   at System.Runtime.Loader.AssemblyLoadContext.ResolveUsingEvent(AssemblyName assemblyName)
   at System.Runtime.Loader.AssemblyLoadContext.ResolveUsingResolvingEvent(IntPtr gchManagedAssemblyLoadContext, AssemblyName assemblyName)
   at System.Reflection.RuntimeAssembly._nLoad(AssemblyName fileName, String codeBase, Evidence assemblySecurity, RuntimeAssembly locationHint, StackCrawlMark& stackMark, IntPtr pPrivHostBinder, Boolean throwOnFileNotFound, Boolean forIntrospection, Boolean suppressSecurityChecks, IntPtr ptrLoadContextBinder)
   at System.Reflection.RuntimeAssembly.InternalGetSatelliteAssembly(String name, CultureInfo culture, Version version, Boolean throwOnFileNotFound, StackCrawlMark& stackMark)
   at System.Resources.ManifestBasedResourceGroveler.GetSatelliteAssembly(CultureInfo lookForCulture, StackCrawlMark& stackMark)
   at System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo culture, Dictionary`2 localResourceSets, Boolean tryParents, Boolean createIfNotExists, StackCrawlMark& stackMark)
   at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo requestedCulture, Boolean createIfNotExists, Boolean tryParents, StackCrawlMark& stackMark)
   at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents)
   at System.Resources.ResourceManager.GetString(String name, CultureInfo culture)
   at System.SR.GetResourceString(String resourceKey, String defaultString)
   at System.Runtime.Serialization.CollectionDataContract.IsCollectionOrTryCreate(Type type, Boolean tryCreate, DataContract& dataContract, Type& itemType, Boolean constructorRequired)
   at System.Runtime.Serialization.DataContract.DataContractCriticalHelper.CreateDataContract(Type type)
   at System.Runtime.Serialization.DataContract.DataContractCriticalHelper.CreateDataContract(Int32 id, RuntimeTypeHandle typeHandle, Type type)
   at System.Runtime.Serialization.DataContract.DataContractCriticalHelper.GetDataContractSkipValidation(Int32 id, RuntimeTypeHandle typeHandle, Type type)
   at System.Runtime.Serialization.DataContract.GetDataContract(RuntimeTypeHandle typeHandle, Type type, SerializationMode mode)
   at System.Runtime.Serialization.Json.DataContractJsonSerializerImpl.get_RootContract()
   at System.Runtime.Serialization.Json.DataContractJsonSerializerImpl.InternalIsStartObject(XmlReaderDelegator reader)
   at System.Runtime.Serialization.Json.DataContractJsonSerializerImpl.InternalReadObject(XmlReaderDelegator xmlReader, Boolean verifyObjectName)
   at System.Runtime.Serialization.XmlObjectSerializer.InternalReadObject(XmlReaderDelegator reader, Boolean verifyObjectName, DataContractResolver dataContractResolver)
   at System.Runtime.Serialization.XmlObjectSerializer.ReadObjectHandleExceptions(XmlReaderDelegator reader, Boolean verifyObjectName, DataContractResolver dataContractResolver)
   at System.Runtime.Serialization.Json.DataContractJsonSerializerImpl.ReadObject(XmlDictionaryReader reader)
   at System.Runtime.Serialization.Json.DataContractJsonSerializerImpl.ReadObject(Stream stream)
   at Gauge.Dotnet.StepExecutor.GetTable(String jsonString) in /Users/vagrant/go-agent-17.4.0/pipelines/dotnet-package/src/StepExecutor.cs:line 84
   at Gauge.Dotnet.StepExecutor.<Execute>b__2_0(String o) in /Users/vagrant/go-agent-17.4.0/pipelines/dotnet-package/src/StepExecutor.cs:line 54
   at System.Linq.Enumerable.SelectArrayIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Gauge.Dotnet.StepExecutor.Execute(GaugeMethod gaugeMethod, String[] args) in /Users/vagrant/go-agent-17.4.0/pipelines/dotnet-package/src/StepExecutor.cs:line 45
   at Gauge.Dotnet.ExecutionOrchestrator.ExecuteStep(GaugeMethod method, String[] args) in /Users/vagrant/go-agent-17.4.0/pipelines/dotnet-package/src/ExecutionOrchestrator.cs:line 57
   at Gauge.Dotnet.Processors.ExecuteStepProcessor.Process(Message request) in /Users/vagrant/go-agent-17.4.0/pipelines/dotnet-package/src/Processors/ExecuteStepProcessor.cs:line 63
   at Gauge.Dotnet.GaugeListener.PollForMessages() in /Users/vagrant/go-agent-17.4.0/pipelines/dotnet-package/src/GaugeListener.cs:line 56
   at Gauge.Dotnet.StartCommand.Execute() in /Users/vagrant/go-agent-17.4.0/pipelines/dotnet-package/src/StartCommand.cs:line 54
   at Gauge.Dotnet.Program.Main(String[] args) in /Users/vagrant/go-agent-17.4.0/pipelines/dotnet-package/src/Program.cs:line 29

        Error Message: write tcp 127.0.0.1:50356->127.0.0.1:50362: use of closed network connection
        Stacktrace:


        Failed Step: Vowels in English language are "aeiou".
        Specification: specs\example.spec:11
        Error Message: Connection closed [127.0.0.1:50362] cause: read tcp 127.0.0.1:50356->127.0.0.1:50362: wsarecv: Eine vorhandene Verbindung wurde vom Remotehost geschlossen.
        Stacktrace:


    Error Message: write tcp 127.0.0.1:50356->127.0.0.1:50362: use of closed network connection
    Stacktrace:

Failed to initialize scenario datastore. Error: write tcp 127.0.0.1:50356->127.0.0.1:50362: use of closed network connection
  ## Vowel counts in multiple word
    Error Message: write tcp 127.0.0.1:50356->127.0.0.1:50362: use of closed network connection
    Stacktrace:



    Error Message: write tcp 127.0.0.1:50356->127.0.0.1:50362: use of closed network connection
    Stacktrace:

Successfully generated html-report to => C:\Dev\Versuche\gauge\core\reports\html-report\index.html
Specifications: 1 executed      0 passed        1 failed        0 skipped
Scenarios:      1 executed      0 passed        1 failed        1 skipped
```

### Steps to reproduce
1. gauge init dotnet
2. gauge run

A similar problem occurs with the csharp template.

### Gauge version
```
Gauge version: 1.0.0
Commit Hash: 5a99965

Plugins
-------
csharp (0.10.4)
dotnet (0.1.0)
html-report (4.0.4)
js (2.3.1)
screenshot (0.0.1)
```

### Additional Versions
Windows 10 1803

**Comment by kuechlerm** [comment]
However, doing the same steps on the Windows Subsystem for Linux on the same machine works ...

**Comment by BugDiver** [comment]
@kuechlerm I'm not able to reproduce your issue at my end. The execution works as expected.
Your issue might be related to this dotnet issue https://github.com/dotnet/corefx/issues/23608.
You can also try to upgrade the dotnet core to latest and see if that works.
If you have any further steps or details to replicate this, please feel free to update here.

**Comment by kuechlerm** [comment]
@BugDiver Thanks for checking.
I think there was a mixture of versions, a mistake on my side and some strange error messages.

Yesterday I updated Visual Studio 2017 (and a million little things with it) now:
- the default gauge dotnet template works
- a second project had a different error about EOF problems

I changed the TargetFramework of the second project to netcoreapp2.1 which resulted in the errors mentioned above. Changing the TargetFramework back to netstandard2.0 fixed it.

I would be great, if setting to TargetFramework to netcoreapp2.1 was possible on Windows - it works on the Windows Subsystem for Linux.

Why the dotnet template suddenly works is beyond me.
